### DISCUSSION TITLE
Scope of the wasmCloud Dashboard

### DISCUSSION THREAD

**Discussion by brooksmtownsend** [wasmCloud/wash-archive#817] [General]
# wasmCloud Dashboard, aka the washboard

As of [ADR-0014](https://github.com/wasmCloud/wasmCloud/blob/main/adr/0014-detach-ui-from-host.md), we accepted the detached washboard as the main way to interact with a wasmCloud lattice via a GUI. This ADR primarily focused on technology choices and the decision to have it connect directly to a lattice, rather than alongside a host. I think we made the best decisions here in terms of contributability and usefulness.

## Responsibility of the washboard
I wanted to open this discussion to propose a strawman and attempt to scope the responsibility of the washboard as a project. Dashboards can grow infinitely in complexity when it comes to visualizations and management and I don't want the washboard to be something that is always adding in complexity so it becomes difficult to maintain.

I think the washboard should:
1. Allow a user to view all of the running components and providers in a wasmCloud lattice
2. Allow a user to view all of the applications created in wadm
3. Allow a user to issue imperative commands to start and stop resources manually, effectively serving as a control interface, as a UI alternative to `wash`
4. Be a good example of how to build custom wasmCloud UI's.
5. ..?

I think the washboard shouldn't:
1. Implement authorization logic (e.g. OAuth2 login to see the dashboard) as this is a huge level of complexity
2. Display host level metrics or traces, as prometheus and OTEL tools are going to be better
3. Have an integrated IDE for editing components

## Thoughts?
That's my strawman, what do you all think?

**Comment by lachieh** [comment]
I concur with all of this. I can't think of anything else we'd need to do, much like no one will ever need more than 640kb of RAM (we don't know what we don't know).

I think it's assumed, but I'll state it for the record; these are the goals—the method of completion and the implementation details are open to interpretation.

**Comment by aschrijver** [comment]
It depends a bit on how wasmCloud is used, what the role of the washboard plays in that. I don't know if you have defined persona's, but for the typical App Developer persona the current washboard is nearly there. It should probably get an additional feature here and there, evolving along with wasmCloud functionality.

Another persona are those integrating wasmCloud to give their own twist to Solution Design to their clients. Like Cosmonic is doing. You might call this persona Integrators. To them it is likely that a) they'll use current washboard as handy dev tool, and b) replace washboard with their own UI in their product. For that use case b) ...

Washboard should:

4. Be a good example of how to build custom UI's.

**Reply by brooksmtownsend** [reply]
I really like this and will integrate it into the discussion above 😄
